nifi-framework-bundle/nifi-framework-extensions/nifi-framework-kubernetes-bundle/nifi-framework-kubernetes-state-provider/src/main/java/org/apache/nifi/kubernetes/state/provider/KubernetesConfigMapStateProvider.java

```diff
@@ -154,64 +154,28 @@ public void shutdown() {
+                    final ConfigMap existingConfigMap = kubernetesClient.configMaps().resource(configMap).get();
+                    if (existingConfigMap == null) {
+                        configMapResult = kubernetesClient.configMaps().resource(configMap).create();
                     } else {
-                        configMapCreated = configMapResource.update();
+                        existingConfigMap.setData(configMap.getData());
+                        configMapResult = kubernetesClient.configMaps().resource(existingConfigMap).update();
                     }
-
                     break;
                 } catch (final KubernetesClientException e) {
                     final int returnCode = e.getCode();
-                    if (returnCode == HttpURLConnection.HTTP_NOT_FOUND) {
-                        // A 404 return code indicates that we need to create the resource instead of update it.
-                        // Now, we will attempt to create the resource instead of update it, so we'll reset the attempt counter.
-                        attempt = 0;
-                        create = true;
-                        continue;
-                    }
-
-                    if (returnCode == HttpURLConnection.HTTP_CONFLICT) {
-                        logger.debug("Update conflict detected when setting state for Component ID [{}]. Attempt {} of {}.", componentId, attempt + 1, MAX_UPDATE_ATTEMPTS);
-
-                        if (attempt < MAX_UPDATE_ATTEMPTS - 1) {
-                            final ConfigMap latestConfigMap = kubernetesClient.configMaps()
-                                    .inNamespace(namespace)
-                                    .withName(configMapName)
-                                    .get();
-
-                            if (latestConfigMap != null) {
-                                final ObjectMeta latestMetadata = latestConfigMap.getMetadata();
-                                final String latestResourceVersion = latestMetadata != null ? latestMetadata.getResourceVersion() : null;
-
-                                if (latestResourceVersion != null) {
-                                    configMap.getMetadata().setResourceVersion(latestResourceVersion);
-                                    configMapResource = kubernetesClient.configMaps().resource(configMap);
-                                    logger.debug("Retrying state update for Component ID [{}] with resource version [{}]", componentId, latestResourceVersion);
-                                    continue;
-                                }
-                            }
-                        }
-
-                        throw e;
-                    }
-
-                    if (returnCode >= 500) {
-                        // Server-side error. We should retry, up to some number of attempts.
+                    if (returnCode == HttpURLConnection.HTTP_CONFLICT || returnCode >= 500) {
```

Comment:
Recommend replacing `500` with `HttpURLConnection.HTTP_INTERNAL_ERROR` to align with the use of `HTTP_CONFLICT`.